src/hotspot/os_cpu/linux_aarch64/os_linux_aarch64.cpp

```diff
@@ -348,7 +354,7 @@ void os::print_tos_pc(outputStream *st, const void *context) {
   // Note: it may be unsafe to inspect memory near pc. For example, pc may
   // point to garbage if entry point in an nmethod is corrupted. Leave
   // this at the end, and hope for the best.
-  address pc = os::Posix::ucontext_get_pc(uc);
+  address pc = os::fetch_frame_from_context(uc).pc();
```

Comment:
What is the difference between `ucontext_get_pc(uc)` and getting the pc from the frame from the uc? How many pc's can a ucontext have?

Reply:
The former returns the PC where the crash happened, which could be in unreadable memory.  Blindly inspecting that memory will cause error reporting to crash.  The latter returns the PC of a frame constructed from the context, and _possibly adjusted to reflect where the caller was when it made the bad call_, which is more useful for getting stack backtraces and less likely to crash error reporting.  When error reporting crashes we recover, but we don't get the information it was dumping when it crashed.